nrf_modem/doc/ug_nrf_modem_porting_os.rst

```diff
@@ -57,6 +57,28 @@ A blind return value of 0 will make all Modem library operations always blocking
 * Report back the remaining time of the timer if the specific timer is interrupted.
 * If timed out, report NRF_ETIMEDOUT.
 
+nrf_modem_os_alloc
+~~~~~~~~~~~~~~~~~~
+
+This function is called by the library to allocate memory dynamically, similarly to a `malloc` call.
+There are no specific requirements as to where to allocate this memory in RAM.
+
+nrf_modem_os_free
+~~~~~~~~~~~~~~~~~
+
+This function should free the memory allocated by `nrf_modem_os_alloc`.
+
+nrf_modem_os_shm_tx_alloc
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This function is called by the library to dynamically allocate memory *to be shared with modem core*.
+This function shall allocate memory on the TX memory region that is passed to the function in :c:func:`nrf_modem_init()` during initialization.
```

Comment:
As you mention that there is no location requirements for the nrf_modem_os_alloc, maybe you can mention here again that this must be in the lower half of the memory?